File: server/src/main/java/com/fluffy/oauth2/infra/github/dto/GithubUserInfoResponse.java
package com.fluffy.oauth2.infra.github.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import com.fluffy.oauth2.domain.OAuth2UserInfo;
import jakarta.annotation.Nullable;

@JsonNaming(SnakeCaseStrategy.class)
public record GithubUserInfoResponse(
        Long id,
        String login,
        String avatarUrl,
        @Nullable String email
) {

    public OAuth2UserInfo toOAuth2UserInfo() {
        return new OAuth2UserInfo(
                id,
                login,
                email,
                avatarUrl
        );
    }
}

File: server/src/main/kotlin/com/fluffy/oauth2/infra/github/response/GithubUserInfoResponse.kt
package com.fluffy.oauth2.infra.github.response

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import com.fluffy.oauth2.domain.OAuth2UserInfo

@JsonNaming(SnakeCaseStrategy::class)
data class GithubUserInfoResponse(
    val id: String,
    val login: String,
    val avatarUrl: String,
    val email: String?,
) {

    fun toOAuth2UserInfo(): OAuth2UserInfo {
        return OAuth2UserInfo(
            id,
            login,
            email,
            avatarUrl,
        )
    }
}